README.md

```diff
@@ -190,3 +191,47 @@ optional arguments:
   -h, --help        show this help message and exit
   --an-integer INT
 ```
+
+## Nested Dataclasses
+
+Dataclasses may be nested; the type of a dataclass field may be another dataclass type:
+
+```python
+from dataclasses import dataclass, field
+
+@dataclass
+class Config:
+  field_a: int
+  field_b: str = "abc"
+
+
+@dataclass
+class Base:
+  config: Config
+  verbose: bool = False
+```
+
+Argument names of fields of the nested dataclass are prefixed with the field name of the nested dataclass in the base
+dataclass. Calling `pydargs.parse(Base, ["-h"])` will result in somthing like:
+
+```text
+usage: your_program.py [-h] --config-field-a CONFIG_FIELD_A
+                            [--config-field-b CONFIG_FIELD_B]
+                            [--verbose VERBOSE]
+
+options:
+  -h, --help            show this help message and exit
+  --verbose VERBOSE     (default: False)
+
+config:
+  --config-field-a CONFIG_FIELD_A
+  --config-field-b CONFIG_FIELD_B
+                        (default: abc)
+
+```
+
+Please be aware of the following:
+- The default (factory) of fields with a dataclass type is ignored by pydargs, which may yield unexpected results.
+  If you must add a default, do `field_name: NestedClass = field(default_factory: NestedClass)`.
+- Nested dataclasses can not be positional (although _fields of_ the nested dataclass can be).
+- Argument names must not collide. In the example above, the `Base` class should not contain a field named `config_field_a`.
```

Comment:
Maybe also write about the behaviour of what happens when this is done. 
Based on the tests it looks like this is not possible and will crash, but it's not immediately clear

Reply:
In that case you'll get an `ArgumentError` indeed. Does the user need to know that? They shouldn't get in that situation anyway.

Reply:
```suggestion
- Argument names must not collide. In the example above, the `Base` class cannot contain a field named `config_field_a`.
```

Reply:
I deliberately chose the wording "should not" as it is not recommended, but one could in principle have a field named `config_field_a` with `ignore_arg=True` to prevent collisions.

See [RFC2119](https://www.rfc-editor.org/rfc/pdfrfc/rfc2119.txt.pdf).

Reply:
Ahhhhh

Reply:
I think this is a good example to add to the docstring though, I didn't think of that

Reply:
Something like: "It is possible, when setting `ignore_arg` to True, but not recommended."

Reply:
Do we really want to go into such detail? The wording "should not" should tell the user that in principle it is a bad idea, unless they understand the full implications of doing so. For the purpose of a README that should be enough, right?

Reply:
👍